src/wp-includes/media.php

```diff
@@ -1751,32 +1756,40 @@ function wp_filter_content_tags( $content, $context = null ) {
 		$context = current_filter();
 	}
 
-	$add_loading_attr = wp_lazy_loading_enabled( 'img', $context );
+	$add_img_loading_attr    = wp_lazy_loading_enabled( 'img', $context );
+	$add_iframe_loading_attr = wp_lazy_loading_enabled( 'iframe', $context );
 
-	if ( false === strpos( $content, '<img' ) ) {
-		return $content;
-	}
-
-	if ( ! preg_match_all( '/<img\s[^>]+>/', $content, $matches ) ) {
+	if ( ! preg_match_all( '/<(img|iframe)\s[^>]+>/', $content, $matches, PREG_SET_ORDER ) ) {
 		return $content;
 	}
 
 	// List of the unique `img` tags found in $content.
 	$images = array();
 
-	foreach ( $matches[0] as $image ) {
-		if ( preg_match( '/wp-image-([0-9]+)/i', $image, $class_id ) ) {
-			$attachment_id = absint( $class_id[1] );
+	// List of the unique `iframe` tags found in $content.
+	$iframes = array();
 
-			if ( $attachment_id ) {
-				// If exactly the same image tag is used more than once, overwrite it.
-				// All identical tags will be replaced later with 'str_replace()'.
-				$images[ $image ] = $attachment_id;
-				continue;
-			}
-		}
+	foreach ( $matches as $match ) {
+		list( $tag, $tag_name ) = $match;
+
+		switch ( $tag_name ) {
```

Comment:
Wondering if this could be better as `if / else` or even `if / elseif`? Will try to do some "speed testing" to see if any difference.

Reply:
I did some speed testing here comparing the use of `switch` with `if / else`, using post content with 25000 images and 25000 iframes for testing. There was no reliably measurable difference in performance, both ended up around 40ms in each run.

Given that we can ignore performance for this decision, I think `switch` is the more appropriate use here. `if / else` would make sense for a true "yes or no" check, but this function is designed to cover various HTML tags, and the list could be expanded in the future. Using a `switch` makes that obvious, and I'd also argue it makes it easier to read (if we were using `else`, it would be immediately clear that `else` applies to `iframe` tags for example).

Reply:
Great! 

> I think switch is the more appropriate use here. if / else would make sense for a true "yes or no" check

Right, should be either `switch` or `if` / `elseif`. As speed is not a factor, it's more of a "personal coding preference". Both work well. The only difference is that `switch` does [loose comparison](https://www.php.net/manual/en/control-structures.switch.php) and we can do strict comparison with `if` / `elseif`, but since we're comparing strings don't think that makes a difference in this case.